promunifi/clients.go

```diff
@@ -0,0 +1,137 @@
+	RxRate            *prometheus.Desc
+	Signal            *prometheus.Desc
+	TxBytes           *prometheus.Desc
+	TxBytesR          *prometheus.Desc
+	TxPackets         *prometheus.Desc
+	TxPower           *prometheus.Desc
+	TxRate            *prometheus.Desc
+	Uptime            *prometheus.Desc
+	WifiTxAttempts    *prometheus.Desc
+	WiredRxBytes      *prometheus.Desc
+	WiredRxBytesR     *prometheus.Desc
+	WiredRxPackets    *prometheus.Desc
+	WiredTxBytes      *prometheus.Desc
+	WiredTxBytesR     *prometheus.Desc
+	WiredTxPackets    *prometheus.Desc
+	DpiStatsApp       *prometheus.Desc
+	DpiStatsCat       *prometheus.Desc
+	DpiStatsRxBytes   *prometheus.Desc
+	DpiStatsRxPackets *prometheus.Desc
+	DpiStatsTxBytes   *prometheus.Desc
+	DpiStatsTxPackets *prometheus.Desc
+}
+
+func descClient(ns string) *uclient {
+	if ns += "_client_"; ns == "_client_" {
+		ns = "client_"
+	}
+
+	labels := []string{"name", "mac", "site_name", "gw_mac", "gw_name", "sw_mac", "sw_name", "vlan", "ip", "oui", "network"}
+	labelWired := append([]string{"sw_port"}, labels...)
+	labelWireless := append([]string{"ap_mac", "ap_name", "radio_name", "radio", "radio_proto", "channel", "essid", "bssid", "radio_desc"}, labels...)
+	wireless := prometheus.Labels{"wired": "false"}
+	wired := prometheus.Labels{"wired": "true"}
+
+	return &uclient{
+		Anomalies:      prometheus.NewDesc(ns+"anomalies_total", "Client Anomalies", labelWireless, wireless),
+		BytesR:         prometheus.NewDesc(ns+"transfer_rate_bytes", "Client Data Rate", labelWireless, wireless),
+		CCQ:            prometheus.NewDesc(ns+"ccq_percent", "Client Connection Quality", labelWireless, wireless),
+		Noise:          prometheus.NewDesc(ns+"noise_db", "Client AP Noise", labelWireless, wireless),
+		RoamCount:      prometheus.NewDesc(ns+"roam_count_total", "Client Roam Counter", labelWireless, wireless),
+		RSSI:           prometheus.NewDesc(ns+"rssi", "Client RSSI", labelWireless, wireless),
+		RxBytes:        prometheus.NewDesc(ns+"receive_bytes_total", "Client Receive Bytes", labelWireless, wireless),
+		RxBytesR:       prometheus.NewDesc(ns+"receive_rate_bytes", "Client Receive Data Rate", labelWireless, wireless),
+		RxPackets:      prometheus.NewDesc(ns+"receive_packets_total", "Client Receive Packets", labelWireless, wireless),
+		RxRate:         prometheus.NewDesc(ns+"radio_receive_rate", "Client Receive Rate", labelWireless, wireless),
+		Signal:         prometheus.NewDesc(ns+"radio_signal", "Client Signal Strength", labelWireless, wireless),
+		TxBytes:        prometheus.NewDesc(ns+"transmit_bytes_total", "Client Transmit Bytes", labelWireless, wireless),
+		TxBytesR:       prometheus.NewDesc(ns+"transmit_rate_bytes", "Client Transmit Data Rate", labelWireless, wireless),
+		TxPackets:      prometheus.NewDesc(ns+"transmit_packets_total", "Client Transmit Packets", labelWireless, wireless),
+		TxPower:        prometheus.NewDesc(ns+"radio_transmit_power", "Client Transmit Power", labelWireless, wireless),
```

Comment:
Also maybe `_db`?

Reply:
I don't know if it's dD. All mine say `22`.

Reply:
Ahh, right, I checked the Unifi UI, it's `dBm`.

Reply:
```suggestion
		TxPower:        prometheus.NewDesc(ns+"radio_transmit_power_dbm", "Client Transmit Power", labelWireless, wireless),
```